students/piechowski_michal/lesson_10_organizing_files/adding_gaps.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import re
+
+dir_path = "test_tree2/"
+prefix = "spam"
+sufix = ".txt"
+path_format = dir_path + prefix + "*" + sufix
+enumerate_pattern = re.compile(r'.*(\d{3})' + sufix)
+
+files_dictionary = {}
+
+
+def fill_files_dictionary():
+    for path in glob.glob(path_format, recursive=True):
+        if enumerate_pattern.match(path):
+            id = int(enumerate_pattern.match(path).group(1))
+            files_dictionary[id] = path
+
+
+def print_files(sorted_ids):
+    for i in sorted_ids:
+        print("File id: " + str(i))
+
+
+def add_gap(sorted_ids):
+    gap = int(input("\nChoose on of above ids to add gap: "))
+    if gap not in sorted_ids:
+        print("Such id doesn't exist")
+        exit()
+
+    for i in reversed(sorted_ids):
+        path = files_dictionary[i]
+        destination = dir_path + prefix + str(i + 1).zfill(3) + sufix
+        os.rename(path, destination)
+        if i == gap:
+            break
+
+
+def main():
+    fill_files_dictionary()
+    sorted_ids = sorted(files_dictionary.keys())
+
+    print_files(sorted_ids)
+    add_gap(sorted_ids)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Tak dla pewności. W tym miejscu można wywołać dowolnie nazwane funkcje i fragmenty poprawnego kodu, nie musi się ona nazywać main (w Javie jest chyba taka funkcja).

Reply:
Tak wiem, że tam może być dowolna funkcja, a nawet nie musi to być funkcja. Po prostu robiąc zadania zaczynam od skopiowania jakiegoś poprzedniego, żeby nie pisać tego "__main__" w kółko i już zostaję przy funkcji main(). Co do czasu oczekiwania to nie ma problemu ;)